pkg/vcs/git.go

```diff
@@ -619,3 +619,11 @@ func (git *git) MergeBases(firstCommit, secondCommit string) ([]*Commit, error)
 	}
 	return ret, nil
 }
+
+func (git *git) CommitExists(commit string) (bool, error) {
+	output, err := git.git("cat-file", "-t", commit)
+	if err != nil && !strings.Contains(err.Error(), "fatal: git cat-file: could not get object info") {
+		return false, fmt.Errorf("failed calling 'git cat-file -f %s': %w", commit, err)
+	}
+	return strings.HasPrefix(string(output), "commit"), nil
```

Comment:
Checking substrings is hacky. We don't know if this particular error string is printed by all current and future git version.
We need to rely on exit status.
I think both `git log -n 1` and `git show` on the commit should return proper exit status.

Reply:
$ git help cat-file
...................
-e
           Exit with zero status if <object> exists and is a valid object. If <object> is of an invalid format, exit with non-zero status and emit an error on stderr.
           
Alternatively I can do vcs.Object(commit, ""). The vcs interface will not be changed then.
It runs 'git show ef3de9e86e58ef3c70da9cf673ee44fc6f1006f8:'.
But the problem will be the same - indistinguishable errors.

Let's cat-file and rely on the error code.

Reply:
Checked the "git cat-file -e ef3de9e86e58ef3c70da9cf673ee44fc6f1006f9" locally.
Exit code 0 -> commit exists.
Exit code 1 -> commit doesn't exist.
Exit code 128 -> Taras started the command in a non-git directory.

Reply:
Done.